diff --git a/font-patcher b/font-patcher
index d14729d3..1a1b7f14 100755
--- a/font-patcher
+++ b/font-patcher
@@ -127,6 +127,12 @@ if args.single:
     verboseAdditionalFontNameSuffix += " Mono"
 
 sourceFont = fontforge.open(args.font)
+new_sfnt_names_list = []
+for x in sourceFont.sfnt_names:
+    if x[0] != 'Japanese':
+        #print x
+        new_sfnt_names_list.append(x)
+sourceFont.sfnt_names = new_sfnt_names_list
 
 # basically split the font name around the dash "-" to get the fontname and the style (e.g. Bold)
 # this does not seem very reliable so only use the style here as a fallback if the font does not
@@ -154,7 +160,7 @@ except IndexError:
     subFamily = fallbackStyle
 
 # some fonts have inaccurate 'SubFamily', if it is Regular let us trust the filename more:
-if subFamily == "Regular":
+if subFamily == "Regular" or subFamily == "regular":
     subFamily = fallbackStyle
 
 if args.windows:
@@ -196,12 +202,22 @@ make_sure_path_exists(args.outputdir)
 
 # comply with SIL Open Font License (OFL)
 reservedFontNameReplacements = {
+    ' '       : '',
+    'NerdFontC': '',
+    'NerdFont': '',
+    'Complete': '',
+    '-book'   : '',
+    'MyricaM' : 'Myrica',
+    'onospace' : '',
+    'VL' : 'VLGothic',
+    'VLGothicGothic' : 'VLGothic',
+    'VLGothicRegular' : 'VLGothic',
+    'RobotoJ Mono' : 'RobotoJ',
+    'RobotoJ-' : 'RobotoJ',
     'source'  : 'sauce',
     'Source'  : 'Sauce',
     'hermit'  : 'hurmit',
     'Hermit'  : 'Hurmit',
-    'fira'    : 'fura',
-    'Fira'    : 'Fura',
     'hack'    : 'knack',
     'Hack'    : 'Knack',
     'hasklig' : 'hasklug',
