diff --git a/font-patcher b/font-patcher
index 95a9634b..31bcfcc2 100755
--- a/font-patcher
+++ b/font-patcher
@@ -151,6 +151,12 @@ if args.single:
     verboseAdditionalFontNameSuffix += " Mono"
 
 sourceFont = fontforge.open(args.font)
+new_sfnt_names_list = []
+for x in sourceFont.sfnt_names:
+    if x[0] != 'Japanese':
+        #print x
+        new_sfnt_names_list.append(x)
+sourceFont.sfnt_names = new_sfnt_names_list
 
 # let's deal with ligatures (mostly for monospaced fonts)
 if args.configfile and config.read(args.configfile):
@@ -195,7 +201,7 @@ except IndexError:
     subFamily = fallbackStyle
 
 # some fonts have inaccurate 'SubFamily', if it is Regular let us trust the filename more:
-if subFamily == "Regular":
+if subFamily == "Regular" or subFamily == "regular":
     subFamily = fallbackStyle
 
 if args.windows:
@@ -237,12 +243,32 @@ make_sure_path_exists(args.outputdir)
 
 # comply with SIL Open Font License (OFL)
 reservedFontNameReplacements = {
+    ' '       : '',
+    'NerdFontC': '',
+    'NerdFont': '',
+    'Complete': '',
+    'MyricaM' : 'Myrica',
+    ' Monospace'     : '-Book',
+    'VL'      : 'VLGothic',
+    'VLGothicGothic' : 'VLGothic',
+    'VLGothicRegular': 'VLGothic',
+    'RobotoJ Mono'   : 'RobotoJ',
+    #'RobotoJ-': 'RobotoJ',
+    'migu'    : 'Migu1M',
+    ' Bold'   : '-Bold',
+    ' Regular': '-Regular',
+    ' Italic':  '-Italic',
+    'RictyOblique': 'Ricty-Oblique',
+    'RictyDiscordOblique': 'RictyDiscord-Oblique',
+    'Ricty Diminished Oblique': 'RictyDiminished-Oblique',
+    'RictyDiminishedDiscordOblique': 'RictyDiminishedDiscord-Oblique',
+    'SarasaMonoJ': 'Sarasa',
+    'SarasaMonoJItalic': 'Sarasa-Italic',
+    'regular' : 'Regular',
     'source'  : 'sauce',
     'Source'  : 'Sauce',
     'hermit'  : 'hurmit',
     'Hermit'  : 'Hurmit',
-    'fira'    : 'fura',
-    'Fira'    : 'Fura',
     'hasklig' : 'hasklug',
     'Hasklig' : 'Hasklug',
     'Share'   : 'Shure',
